src/deprecated.jl

```diff
@@ -0,0 +1,12 @@
+@deprecate(termnames(C::AbstractContrasts, levels::AbstractVector, baseind::Integer),
+           coefnames(C::AbstractContrasts, levels::AbstractVector, baseind::Integer),
+           false)
+
+function Base.getproperty(cm::ContrastsMatrix, x::Symbol)
+    if x === :termnames 
+        Base.depwarn("The `termnames` field has been renamed `coefnames`.", :ContrastsMatrix)
+        x = :coefnames
+    end
+
+    return getfield(cm, x)
```

Comment:
```suggestion
    if x === :termnames
        Base.depwarn("the `termnames` field of `ConstrastsMatrix` is deprecated; use `coefnames(cm)` instead."
        return coefnames(cm)
    else
        return getfield(cm, x)
    end
```
Might as well make the deprecation target encourage better behavior